class/practice2.1/AdvancedHelloWorld.cpp

```diff
@@ -28,67 +57,46 @@ class Database {
         }
         file.close();
     }
- std::string deleteUser(const std::string& nameToFind) {
-    auto name = statsData.find(nameToFind);
-    if (name != statsData.end()) {
-        statsData.erase(name);
-    }
-    SaveData();
-    return "All data about this user was cleared";
-}
-
-    std::string UserIncrement(const std::string& nameToFind) {
-        std::string returnValue;
-        if (statsData.find(nameToFind)!=statsData.end())
-        {
-            statsData [nameToFind]++;
-            returnValue="Hello again(x" + std::to_string(statsData[nameToFind]) + ")," + nameToFind;
-        }else{
-            statsData [nameToFind]++;
-            returnValue="Welcome " +nameToFind;
-        }
-        SaveData();
-        return returnValue;
-    }
-public:
-    Database(const std::string& filePath) : database(filePath) {
-         LoadData();
-    }
-    void clearDatabase() {
-        statsData.clear();
-        SaveData();
-    }
-    std::string PerformChanges(const std::string& nameToFind, const std::string& additionalCommand) {
-        if (statsData.find(nameToFind)==statsData.end()) {
-            if (additionalCommand == deleteCommand) {
-               return "There is no such user ";
-            }
-            return UserIncrement(nameToFind);
-        }
-        else {
-            if (additionalCommand == deleteCommand) {
-                return deleteUser(nameToFind);
-            }
-            return UserIncrement(nameToFind);
-        }
-    }
 };
 int main(int argc, char* argv[])
 {
     Database db("database.txt");
+
     if (argc < 2) {
         std::cout << "Please provide a name to find." << std::endl;
-        return 0;
+        return 1;
     }
+
     std::string nameToFind;
     nameToFind = argv[1];
     const auto additionalCommand = std::string(argc > 2 ? argv[2] : "");
+
     if (nameToFind == bread)
     {
         db.clearDatabase();
         std::cout << "all data was cleared" << std::endl;
         return 0;
     }
-   std::cout << db.PerformChanges(nameToFind,additionalCommand)<< std::endl;
+
+    std::map<std::string, int> statsData=db.getStatsData();
```

Comment:
map copy is a big overhead and encapsulation is broken with it too.
The real intent of your code is to check the fact whether userName is present in database or not - it can be done with a class method and will save a lot of resources and will not break encapsulation.